pkg/ui/src/components/badge/badge.styl

```diff
@@ -0,0 +1,58 @@
+// Copyright 2019 The Cockroach Authors.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.txt.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0, included in the file
+// licenses/APL.txt.
+
+@require '~src/components/core/index.styl'
+
+.badge
+  display flex
+  flex-direction row
+  border-radius 3px
+  text-transform uppercase
+  width fit-content
+  padding $spacing-xx-small $spacing-x-small
+  cursor default
+
+.badge--size-medium
+  height $line-height--medium
+  font-size $font-size--small
+  font-weight $font-weight--medium
+  line-height $line-height--xx-small
+  letter-spacing 1.5px
+
+.badge--size-small
```

Comment:
Exactly, empty classes are ordered in a way so they fall to default classes `.badge__icon` and `.badge--status-default`.
It does allow for now pay more attention to the functionality required in Admin UI rather than on the comprehensive implementation of components.

Here are compiled styles to CSS (identical to what you suggested :) 
![Screenshot 2019-12-19 at 10 58 17](https://user-images.githubusercontent.com/3106437/71159728-dedc6280-224e-11ea-888f-e320b49dbbad.png)